### PR TITLE
address associations pallet - demo

### PR DESCRIPTION
**Pull request by AmbientTea**
# Description

This is a demo of the client-facing API of the address associations pallet.

### CODE REVIEW

File: pallets/address-associations/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+	pub const PALLET_VERSION: u32 = 1;
+
+	#[pallet::pallet]
+	pub struct Pallet<T>(_);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		/// Main chain address type
+		type MainChainAddress: Member + Parameter + MaxEncodedLen;
```
Comment (LGLO) [inline pallets/address-associations/src/lib.rs:23]: I would make it a `Key`

File: pallets/address-associations/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+		/// Partner chain address type
+		type PartnerChainAddress: Member + Parameter + MaxEncodedLen;
```
Comment (LGLO) [inline pallets/address-associations/src/lib.rs:26]: I would make it a `Value` and make the pallet cardano-associations. I don't even know if cardano is important part here.

File: pallets/address-associations/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+	impl<T: Config> Pallet<T> {
+		/// Changes the observability configuration.
+		///
+		/// This extrinsic must be run either using `sudo` or some other chain governance mechanism.
+		///
+		/// Warning: this extrinsic also resets the internal state stored by the observability data source
+		/// in the runtime and should be invoked only when necessary.
+		#[pallet::call_index(1)]
+		#[pallet::weight((1, DispatchClass::Normal))]
+		pub fn set_main_chain_scripts(
```
Comment (LGLO) [inline pallets/address-associations/src/lib.rs:64]: I propose `set_observability_configuration`.

File: pallets/address-associations/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+	}
+
+	impl<T: Config> Pallet<T> {
+		/// Returns the current pallet version.
+		pub fn get_version() -> u32 {
+			PALLET_VERSION
+		}
+
+		/// Retrieves all main chain - partner chain address associations from the runtime storage.
+		pub fn get_all_address_associations() -> Vec<(T::MainChainAddress, T::PartnerChainAddress)>
```
[the remark this answers is not in the document]
Comment (AmbientTea) [reply pallets/address-associations/src/lib.rs:82]: In principle they are. As Midnight or any other chain grows, there will be more and more associations. Also, there's no de-association, so they stay forever. Perhaps this function won't be needed at all, or we could return `impl Iterator`.

File: pallets/address-associations/src/lib.rs (hunk 2 above)
Comment (AmbientTea) [reply pallets/address-associations/src/lib.rs:26]: I don't think Cardano is important here, but having a `key-pair-pallet` would be too general. I'd like to stay within the domain "associating some main chain identifier with some partner chain identifier", whatever chain the main chain and partner chain happen to be.

**Pull request closed without merging**